src/java.base/share/classes/jdk/internal/vm/ThreadSnapshot.java

```diff
@@ -52,12 +52,14 @@ private ThreadSnapshot() {}
 
     /**
      * Take a snapshot of a Thread to get all information about the thread.
+     * Return null if a ThreadSnapshot cannot be created, for example if the
+     * thread has terminated.
```

Comment:
Strictly speaking we can create a snapshot for a terminated thread, but our only client doesn't care about them so we don't.